packages/animation/src/extendAnimator/extendAnimator.test.tsx

```diff
@@ -31,12 +31,15 @@ test('Should extend actual animated component', () => {
 
 test('Should extend class animator settings and filter unknown settings for animated component', () => {
   let animator: any;
-  const ExampleComponent: FC<any> = props => {
+  interface ExampleAnimator {
+    animator?: AnimatorSettings
```

Comment:
A wrapped component with `withAnimator` will receive the `animator` prop as `AnimatorProvidedSettings` interface.

Interface `AnimatorClassSettings` is the settings allowed for a component "class" definition. With "class" I don't mean the OOP class, but the common settings for all the instance components created with it. It is used in `withAnimator(classSettings: AnimatorClassSettings)` or `extendAnimator(classSettings: AnimatorClassSettings)`.

Interface `AnimatorInstanceSettings` is the settings allowed for a component instance when it is instantiated in the React tree. So these settings should only be allowed to that component itself, not shared. It is used in `<MyAnimatedComponent animator={instanceAnimatorSettings} />`.

Interface `AnimatorProvidedSettings` is the settings received inside the wrapped animator component. It is used when `const MyAnimatedComponent = ({ animator }) { ... };` where `animator: AnimatorProvidedSettings`.

Interface `AnimatorSettings` is the final settings which the `Animator` component receives. It is used in `<Animator animator={animatorSettings}>...</Animator>`.

I think this documentation should be in the guides, but I haven't done it yet :sweat_smile: .

Let me know if you have questions or if you have recommendations about it :wink: I wanted to make it simple but I don't know if I overcomplicate it.

Reply:
Aha. I might have gone about this the wrong way then.. I'll take a look later and try and rework what i did.

Reply:
Yes, in this case, it should be as:

```ts
interface ExampleExtendAnimatorProps {
  animator: AnimatorProvidedSettings
}
```

Since that what the component will finally receive. It will always be received so it is not necessary to set it as optional.

You can use the same interface `ExampleExtendAnimatorProps` in line 15.

Everything else looks good to me :wink: